overlord/hookstate/ctlcmd/set.go

```diff
@@ -42,6 +56,51 @@ func (s *setCommand) Execute(args []string) error {
 		return fmt.Errorf("cannot set without a context")
 	}
 
-	// TODO: Talk to the handler to take care of the set request.
+	// Extract the transaction from the context. If none, make one.
+	s.context().Lock()
+	transaction, ok := s.context().Cached("transaction").(*configstate.Transaction)
+	s.context().Unlock()
+	if !ok {
```

Comment:
This needs to be inside the lock, right? Otherwise a race will end up with independent transactions at hand, and we also want to add OnDone only once.